File: mq/php/producer.php
<?php

include 'JsonRPC.php';

$n = 0;
$topics = ['topic_1', 'topic_2', 'topic_3', 'topic_4', 'topic_5'];
while (true) {
    try {
        $rpc = new JsonRPC('tcp://127.0.0.1:9503', 30);
        $id = 'xxxx_id' . microtime(true) . rand(1, 999999999);
        list($res, $err) = $rpc->push([
            'id'    => $id,
            'topic' => $topics[array_rand($topics, 1)],
            'body'  => 'this is a rpc test',
            'delay' => (string)rand(1, 60),
            'TTR'   => (string)rand(0, 30)
        ]);
        if ($err) {
            echo sprintf("err:%s \n", $err->msg);
        } else {
            echo sprintf("%s success \n", $id);
        }

         $n++;
         if ($n > 10000) {
            break;
         }
    } catch (Exception $e) {
        echo $e->getMessage() . PHP_EOL;
        sleep(3);
    }
}

File: mq/python/producer.py
# coding=utf-8
__author__ = 'wuzhc'
'''
client端
'''
import socket
import json
import random
import uuid

host = '127.0.0.1'
port = 9503
client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
client.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
client.connect((host, port))

n = 0
topic = ['topic_1', 'topic_2', 'topic_3', 'topic_4']
while True:
    job = {
        'id': 'xxxx_id' + str(uuid.uuid4()),
        'topic': topic[random.randint(0, len(topic) - 1)],
        'body': 'this is a job',
        'delay': str(random.randint(0, 60)),
        'TTR': str(random.randint(0, 30))
    }
    data = {
        'method': 'Service.Push',
        'params': [job],
        'id': str(uuid.uuid4())
    }
    client.send((json.dumps(data) + '\n').encode())
    res = client.recv(1024)
    print(res.decode())

    n += 1
    if n > 10000:
        break
